includes/pg-functions.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+
+
+/**
+ * Check if logs table exists
+ */
+function isPgLogsTableCreated()
+{
+    global $wpdb;
+    $tableName = $wpdb->prefix . PG_LOGS_TABLE_NAME;
+    if ($wpdb->get_var("SHOW TABLES LIKE '$tableName'") == $tableName) {
+        return true;
+    }
+
+    return false;
+}
+
+
+
+function createPgLogsTable()
+{
+    global $wpdb;
+    $charset_collate = $wpdb->get_charset_collate();
+    $LogsTableName   = $wpdb->prefix . PG_LOGS_TABLE_NAME;
+    $sqlQuery        = "CREATE TABLE $LogsTableName ( 
+    id int NOT NULL AUTO_INCREMENT,
+    log text NOT NULL, 
+    createdAt timestamp DEFAULT CURRENT_TIMESTAMP,
+    UNIQUE KEY id (id)) 
+    $charset_collate";
+    require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+    dbDelta($sqlQuery);
+}
+
+/**
+ * @param null $exception
+ * @param null $message
+ */
+function insertLogEntry($exception = null, $message = null)
+{
+    global $wpdb;
+    if (! isPgLogsTableCreated()) {
+        createPgLogsTable();
+    }
+    $logEntry = new Pagantis\ModuleUtils\Model\Log\LogEntry();
+    if ($exception instanceof \Exception) {
+        $logEntry = $logEntry->error($exception);
+    } else {
+        $logEntry = $logEntry->info($message);
+    }
+    $tableName = $wpdb->prefix . PG_LOGS_TABLE_NAME;
+    $wpdb->insert($tableName, array('log' => $logEntry->toJson()));
+}
+
+/**
+ * @return bool
+ */
+function areDecimalSeparatorEqual()
+{
+    $decimalSeparator                = getPgSimulatorDecimalSeparatorConfig();
+    $wc_get_number_of_price_decimals = get_option('woocommerce_price_decimal_sep');
+    if ($wc_get_number_of_price_decimals == $decimalSeparator) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+
+/**
+ * @return bool
+ */
+function areThousandsSeparatorEqual()
+{
+    $thousandSeparator               = getPgSimulatorThousandsSeparator();
+    $wc_get_price_thousand_separator = get_option('woocommerce_price_thousand_sep');
+
+    if ($wc_get_price_thousand_separator == $thousandSeparator) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * @return array|object|null
+ */
+function getPgSimulatorThousandsSeparator()
+{
+    global $wpdb;
+    $tableName = $wpdb->prefix . PG_CONFIG_TABLE_NAME;
+    $query     = "SELECT * FROM $tableName WHERE config='PAGANTIS_SIMULATOR_THOUSANDS_SEPARATOR'";
+    $result    = $wpdb->get_results($query, ARRAY_A);
+    return $result;
+}
+
+/**
+ * @return array|object|null
+ */
+function getPgSimulatorDecimalSeparatorConfig()
+{
+    global $wpdb;
+    $tableName = $wpdb->prefix . PG_CONFIG_TABLE_NAME;
+    $query     = "SELECT * FROM $tableName WHERE config='PAGANTIS_SIMULATOR_DECIMAL_SEPARATOR'";
```

Comment:
try not to use "select * ..."

Reply:
This method returns an arry/object and is compared with 
```
$wc_get_number_of_price_decimals = get_option('woocommerce_price_decimal_sep');
```

Is correct? scalar values is a better option in this case I think

Reply:
> try not to use "select * ..."

I'm confused about this suggestion

Reply:
> `$wc_get_number_of_price_decimals = get_option('woocommerce_price_decimal_sep');`

get_option('woocommerce_price_decimal_sep') returns  "."
[see test](https://github.com/woocommerce/woocommerce/blob/master/tests/legacy/unit-tests/formatting/functions.php#L551)
I also dumped the value in the wc-logs with my logging class to double check

Reply:
so you're comparin an array with a string? on line 79
```
    $thousandSeparator               = getPgSimulatorThousandsSeparator(); -> returns array/object
    $wc_get_price_thousand_separator = get_option('woocommerce_price_thousand_sep'); -> returns string

     if ($wc_get_price_thousand_separator == $thousandSeparator) {
```

Reply:
This has been resolved [in this commit](https://github.com/pagantis/woocommerce/pull/55/commits/8e2cd90c1b5cada4a12683b2d4205301bc28e1d6)

Reply:
ok!